File: benchmarks/comparison/pmp/example-sqrt3.cpp
#include <pmp/algorithms/SurfaceSubdivision.h>
#include <pmp/SurfaceMesh.h>

using pmp::SurfaceMesh;
using pmp::SurfaceSubdivision;

int main() {
    SurfaceMesh mesh;
    if (!mesh.read("input.stl")) {
        std::cerr << "Error reading file" << std::endl;
        return 1;
    }

    auto sub = SurfaceSubdivision(mesh);
    sub.sqrt3();

    if (!mesh.write("output.ply")) {
        std::cerr << "Error writing file" << std::endl;
        return 1;
    }
}

File: benchmarks/comparison/lox/example-sqrt3.rs
use lox::{
    algo,
    ds::{HalfEdgeMesh, half_edge::TriConfig},
    fat::MiniMesh,
    io,
};

fn main() -> Result<(), io::Error> {
    type MyMesh = MiniMesh<HalfEdgeMesh<TriConfig>>;

    let mut mesh: MyMesh = io::read_file("input.stl")?;

    algo::subdivision::sqrt3(&mut mesh.mesh, &mut mesh.vertex_positions, 1);

    io::write_file(&mesh, "output.ply")?;

    Ok(())
}
